### ISSUE TITLE
yesod-test does not follow redirects

### ISSUE THREAD

**Issue by mwotton** [yesodweb/yesod#1190]
We have some code that posts login details:

```
loginAsUser email password = do
  -- You _must_ get the page or addToken will fail!
  get BTLoginR
  statusIs 200
  request $ do
    setMethod "POST"
    addPostParam "email" email
    addPostParam "password" password
    addPostParam "remember-me" "yes"
    setUrl BTLoginR
    addToken
```

Once we called this code, we expected the token to be set, and addToken to work on the next post.
Unfortunately it doesn't: as far as i can tell, this is because addToken is looking at the previous response to scan for a CSRF token, and the 303 REDIRECT response from /login empty. If yesod-test followed the redirect, or offered some way of following the redirect, we would get the logged-in homepage, and I think everything would be kosher,  but as it doesn't, the next post fails.

Changing it to addTokenFromCookie works, but both should work, I think.
(If I have missed some obvious way to tell yesod-test to follow redirects, please forgive me.)

This is with

```
  yesodMiddleware = defaultCsrfMiddleware . sessionMiddleware . defaultYesodMiddleware
```

set,

```
✗ stack list-dependencies | grep yesod
classy-prelude-yesod 0.12.5
yesod 1.4.2
yesod-auth 1.4.12
yesod-core 1.4.19
yesod-form 1.4.6
yesod-newsfeed 1.5
yesod-persistent 1.4.0.3
yesod-static 1.5.0.3
```

**Comment by snoyberg** [comment]
Following redirects in general wouldn't be good behavior for yesod-test (since we'll often want to test that a route really does return a 30X response). But having an alternate function that follows redirects would seem like a logical addition. Would that address your concern here? Would you be interested in trying out a PR for that?

**Comment by mwotton** [comment]
Yep, that's what I need. Would something like "followLastRedirect" that
looks at the last response and errors out if it wasn't an la redirect be
the right way to approach it?

**Comment by snoyberg** [comment]
Yeah, sounds good

**Comment by MaxGabriel** [comment]
Closing; this is implemented by #1194
